schneiderWallbox/integrationpluginschneiderwallbox.cpp

```diff
@@ -0,0 +1,546 @@
+* contact@nymea.io or see our FAQ/Licensing Information on
+* https://nymea.io/license/faq
+*
+* * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * */
+
+#include "integrationpluginschneiderwallbox.h"
+#include "plugininfo.h"
+
+#include <network/networkdevicediscovery.h>
+#include <types/param.h>
+#include <plugintimer.h>
+
+#include <QDebug>
+#include <QStringList>
+#include <QJsonDocument>
+#include <QNetworkInterface>
+
+
+IntegrationPluginSchneiderWallbox::IntegrationPluginSchneiderWallbox()
+{
+
+}
+
+void IntegrationPluginSchneiderWallbox::discoverThings(ThingDiscoveryInfo *info)
+{
+    if (!hardwareManager()->networkDeviceDiscovery()->available()) {
+        qCWarning(dcSchneiderElectric()) << "Failed to discover network devices. The network device discovery is not available.";
+        info->finish(Thing::ThingErrorHardwareNotAvailable, QT_TR_NOOP("The discovery is not available. Please enter the IP address manually."));
+        return;
+    }
+
+    qCDebug(dcSchneiderElectric()) << "Starting network discovery...";
+    NetworkDeviceDiscoveryReply *discoveryReply = hardwareManager()->networkDeviceDiscovery()->discover();
+    connect(discoveryReply, &NetworkDeviceDiscoveryReply::finished, info, [=](){
+        qCDebug(dcSchneiderElectric()) << "Discovery finished. Found" << discoveryReply->networkDeviceInfos().count() << "devices";
+        foreach (const NetworkDeviceInfo &networkDeviceInfo, discoveryReply->networkDeviceInfos()) {
+            qCDebug(dcSchneiderElectric()) << networkDeviceInfo;
+
+            if (!networkDeviceInfo.macAddressManufacturer().contains("TELEMECANIQUE", Qt::CaseSensitivity::CaseInsensitive)) {
+                continue;
+            }
+
+            ThingClass thingClass = supportedThings().findById(info->thingClassId());
+            ParamTypeId macAddressParamType = thingClass.paramTypes().findByName("mac").id();
+
+            ThingDescriptor descriptor(info->thingClassId(), thingClass.displayName(), networkDeviceInfo.address().toString());
+            descriptor.setParams({Param(macAddressParamType, networkDeviceInfo.macAddress())});
+
+            // Check if we already have set up this device
+            Thing *existingThing = myThings().findByParams(descriptor.params());
+            if (existingThing) {
+                qCDebug(dcSchneiderElectric()) << "Found already existing" << thingClass.name() << "wallbox:" << existingThing->name() << networkDeviceInfo;
+                descriptor.setThingId(existingThing->id());
+            } else {
+                qCDebug(dcSchneiderElectric()) << "Found new" << thingClass.name() << "wallbox";
+            }
+
+            info->addThingDescriptor(descriptor);
+        }
+        info->finish(Thing::ThingErrorNoError);
+    });
+}
+
+
+void IntegrationPluginSchneiderWallbox::setupThing(ThingSetupInfo *info)
+{
+    Thing *thing = info->thing();
+
+    /* Probably not needed.
```

Comment:
Indeed, not needed. if the .json defines only one thing class, you'll never get any other thingclassId here. Anything else would be a bug in nymea core which I though should be caught by the existing auto-tests. You can safely delete this.